### ISSUE TITLE
Is it possible to have a tree made of one leaf only?

### ISSUE THREAD

**Issue by pommedeterresautee** [tqchen/xgboost#141]
I generated a tree which looks like that
```
booster[0]
0:[f45522<1.00001] yes=1,no=2,missing=2
	1:leaf=1.99982
	2:leaf=-1.87302
booster[1]
0:[f45522<1.00001] yes=1,no=2,missing=2
	1:leaf=1.13512
	2:leaf=-1.00598
booster[2]
0:[f45522<1.00001] yes=1,no=2,missing=2
	1:leaf=1.04291
	2:leaf=-0.790276
booster[3]
0:[f45522<1.00001] yes=1,no=2,missing=2
	1:leaf=1.01379
	2:leaf=-0.603541
booster[4]
0:leaf=0.994874
```

I am wondering if it is a bug or if it makes sense in some way I don't understand?

**Comment by tqchen** [comment]
Yes, it is possible. xgboost will choose not to split when splitting do not help you improve gain that much. For correctness check, you need to check training loss convergence

**Comment by pommedeterresautee** [comment]
Ok I understand.

It is ok if I filter out these one leaf tree of model ploting function, importance feature function, am I right? There is nothing to learn from it? Just the sample it gots for the nth round have nothing to teach us, right?

**Comment by pommedeterresautee** [comment]
Tks
